dspace-api/src/main/java/org/dspace/content/MetadataValueServiceImpl.java

```diff
@@ -99,6 +105,21 @@ public Iterator<MetadataValue> findByFieldAndValue(Context context, MetadataFiel
         return metadataValueDAO.findItemValuesByFieldAndValue(context, metadataField, value);
     }
 
+    @Override
+    public List<Pair<DSpaceObject, Integer>> findByFieldAndValue(
+        Context context, Map<String, String> fieldsValues, boolean keepAuthority
+    ) throws SQLException, IllegalArgumentException {
+        Map<Integer, String> fieldsMap = new HashMap<>();
+        // Convert each metadata field string into the corresponding metadata field id.
+        for (Map.Entry<String, String> entry : fieldsValues.entrySet()) {
+            fieldsMap.put(
+                metadataFieldService.findByString(context, entry.getKey(), '.').getID(),
```

Comment:
What's happen if the `MetadataField` is not found. If `findByString()` return null, this will cause an exception